torch/csrc/api/src/nn/modules/embedding.cpp

```diff
@@ -11,27 +12,206 @@
+        "embedding_bag: If per_sample_weights (", (*per_sample_weights).sizes(), ") is not null, then it must have the same shape as the input (", input.sizes(), ")");
+      if(input.dim() == 2) {
+        TORCH_CHECK(offsets == c10::nullopt,
+          "If input is 2D, then offsets has to be null, as input is treated is a mini-batch of fixed length sequences. However, found offsets of type Tensor");
+          offsets = torch::arange(0, input.numel(), input.size(1),
+                                     torch::TensorOptions().dtype(torch::kLong).device(input.device()));
+          reshape_input = true;
+          //input = input.reshape(-1);
+          if(per_sample_weights != c10::nullopt) {
+            per_sample_weights = (*per_sample_weights).reshape(-1);
+          }
+      }
+      else if(input.dim() == 1) {
+        TORCH_CHECK(offsets != c10::nullopt, "offsets has to be a 1D Tensor but got null");
+        TORCH_CHECK((*offsets).dim() == 1, "offsets has to be a 1D Tensor");
+        TORCH_CHECK((*offsets)[0].item<int64_t>() == 0, "offsets[0] has to be 0, i.e., the first sequence in the mini-batch has to start from position 0. However, got ",
+         (*offsets)[0].item<int64_t>());
+        TORCH_CHECK((*offsets)[-1].item<int64_t>() <= input.size(0), "offsets[-1] can not be greater than input's length({)",
+                  input.size(0), "}), but got offsets[-1] of {", (*offsets)[-1].item<int64_t>(), "}");
+      }
+      else{
+        TORCH_CHECK(false, "input has to be 1D or 2D Tensor,but got Tensor of dimension ", input.dim());
+      }
+
+      int mode_enum;
+      if(options.mode() == "sum") {
+        mode_enum = 0;
+      }
+      else if(options.mode() == "mean") {
+        mode_enum = 1;
+      }
+      else if(options.mode() =="max") {
+        mode_enum = 2;
+        TORCH_CHECK(!options.scale_grad_by_freq(), "max mode does not support scaling the gradient by the frequency");
+        TORCH_CHECK(!options.sparse(), "max mode does not support sparse weights");
+      }
+      else{
+        TORCH_CHECK(false, "mode has to be one of sum, mean or max");
+      }
+
+      if(options.max_norm() != c10::nullopt) {
+        torch::NoGradGuard no_grad;
+        if(reshape_input){
+          torch::embedding_renorm_(weight, input.reshape(-1), *options.max_norm(), options.norm_type());
+        }
+        else{
+          torch::embedding_renorm_(weight, input, *options.max_norm(), options.norm_type());
+        }
+      }
+
+      TORCH_CHECK((per_sample_weights == c10::nullopt) || (options.mode() == "sum"), "embedding_bag: per_sample_weights was not null. ",
+            "per_sample_weights is only supported for mode='sum' (got mode='",
+            options.mode(), "').Please open a feature request on GitHub.");
+      if(reshape_input){
+        return torch::embedding_bag(weight, input.reshape(-1), *offsets, options.scale_grad_by_freq(), mode_enum, options.sparse(), *per_sample_weights);
```

Comment:
To match the Python version:
https://github.com/pytorch/pytorch/blob/b0bb5e338eab7034bd112c48675c4b754a7763b5/torch/nn/functional.py#L1632-L1640
we should only return the first element of the tuple here, and change the return type of `EmbeddingBag::forward()` to `torch::Tensor`